src/main/java/codeu/model/store/basic/HashtagStore.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.model.store.basic;
+
+import codeu.model.data.Hashtag;
+import codeu.model.store.persistence.PersistentStorageAgent;
+import java.time.Instant;
+import java.util.Hashtable;
+import java.util.UUID;
+
+public class HashtagStore {
+
+  /** Singleton instance of HashtagStore. */
+  private static HashtagStore instance;
+
+  /**
+   * Returns the singleton instance of HashtagStore that should be shared between all servlet
+   * classes. Do not call this function from a test; use getTestInstance() instead.
+   */
+  public static HashtagStore getInstance() {
+    if (instance == null) {
+      instance = new HashtagStore(PersistentStorageAgent.getInstance());
+    }
+    return instance;
+  }
+
+  /**
+   * Instance getter function used for testing. Supply a mock for PersistentStorageAgent.
+   *
+   * @param persistentStorageAgent a mock used for testing
+   */
+  public static HashtagStore getTestInstance(PersistentStorageAgent persistentStorageAgent) {
+    return new HashtagStore(persistentStorageAgent);
+  }
+
+  /**
+   * The PersistentStorageAgent responsible for loading Users from and saving Users to Datastore.
+   */
+  private PersistentStorageAgent persistentStorageAgent;
+
+  /** The in-memory list of Hashtags. */
+  private Hashtable<String, Hashtag> hashtags;
```

Comment:
Oh, I just noticed this.  You should use HashMap instead of Hashtable.  There's a short explanation here: https://docs.oracle.com/javase/7/docs/api/java/util/Hashtable.html, but basically Hashtable is unnecessarily inefficient and thus not recommended.

Reply:
Indeed! It makes sense, it's changed now